MekHQ/src/mekhq/campaign/stratcon/StratconTrackState.java

```diff
@@ -404,6 +404,32 @@ public void failObjective(StratconCoords coords) {
         }
     }
     
+    /**
+     * Whether or not this track has a facility on it that reveals the track.
+     */
+    public boolean hasActiveTrackReveal() {
+        for (StratconFacility facility : getFacilities().values()) {
+            if (facility.getRevealTrack()) {
+                return true;
+            }
+        }
+        
+        return false;
+    }
+    
+    /**
+     * Count of all the scenario odds adjustments from facilities
+     * (and potentially other sources) on this track.
+     */
+    public int getScenarioOddsAdjustment() {
+        int accumulator = 0;
+        for (StratconFacility facility : getFacilities().values()) {
+            accumulator += facility.getScenarioOddsModifier();
+        }
+        
+        return accumulator;
```

Comment:
Stream implementation would be better here.

Reply:
Still holds.

getFacilities().values().mapToInt(StratconFacility::getScenarioOddsModifier).sum() would be the code to do so